lib/openstack_query/query_server.py

```diff
@@ -0,0 +1,144 @@
+        QueryPresets.EQUAL_TO: ["*"],
+        QueryPresets.NOT_EQUAL_TO: ["*"],
+        QueryPresets.OLDER_THAN: [
+            ServerProperties.SERVER_CREATION_DATE,
+            ServerProperties.SERVER_LAST_UPDATED_DATE,
+        ],
+        QueryPresets.YOUNGER_THAN: [
+            ServerProperties.SERVER_CREATION_DATE,
+            ServerProperties.SERVER_LAST_UPDATED_DATE,
+        ],
+        QueryPresets.YOUNGER_THAN_OR_EQUAL_TO: [
+            ServerProperties.SERVER_CREATION_DATE,
+            ServerProperties.SERVER_LAST_UPDATED_DATE,
+        ],
+        QueryPresets.OLDER_THAN_OR_EQUAL_TO: [
+            ServerProperties.SERVER_CREATION_DATE,
+            ServerProperties.SERVER_LAST_UPDATED_DATE,
+        ],
+        QueryPresets.MATCHES_REGEX: [ServerProperties.SERVER_NAME],
+    }
+
+    _NON_DEFAULT_FILTER_FUNCTION_MAPPINGS = {}
+
+    def __init__(self, connection_cls=OpenstackConnection):
+        QueryWrapper.__init__(self, connection_cls)
+
+    @staticmethod
+    def _run_query(
+        conn: OpenstackConnection,
+        filter_kwargs: Optional[Dict[str, str]] = None,
+        **kwargs
+    ):
+        """
+        This method runs the query by running openstacksdk commands
+        For QueryServer, this command gets all projects available and iteratively finds servers in that project
+        :param conn: An OpenstackConnection object
+        :param filter_kwargs: An Optional set of kwargs to pass to conn.compute.servers
+        :param kwargs: a set of extra meta params that override openstacksdk commands or change which commands are run
+
+        kwargs: For QueryServer:
+            'from_projects' - set which projects to search by providing a list of openstack projects to search in
+        """
+        servers = []
+        if "from_projects" in kwargs:
+            projects = kwargs["from_projects"]
+        else:
+            projects = []
+            for project in conn.identity.projects():
+                projects.append(project)
+
+        for project in projects:
+            servers.extend(
+                QueryServer._run_query_on_project(conn, project, filter_kwargs)
+            )
+        return servers
+
+    @staticmethod
+    def _run_query_on_project(
+        conn: OpenstackConnection,
+        project: Project,
+        filter_kwargs: Optional[Dict[str, str]] = None,
+    ):
+        servers = []
+
+        server_filters = {"project_id": project["id"], "all_tenants": True}
+
+        server_filters.update(filter_kwargs if filter_kwargs else {})
+        servers.extend(list(conn.compute.servers(filters=server_filters)))
+
+        return servers
+
+    @staticmethod
+    def _parse_subset(conn: OpenstackConnection, server_identifiers: Server):
+        query_from = []
+        for server_instance in server_identifiers:
+            server_obj = server_instance
+            if not isinstance(Server, server_instance):
+                server_obj = conn.compute.find_server(server_instance)
```

Comment:
As me and Ramzi discovered, this is a really really expensive operation in Openstack
It's (ironically) faster to grab all the servers from a project and filter project side, than to do the auth flow n times for each machine

Reply:
sure that makes sense :+1:

Reply:
For now the parse_subset function simply accepts whatever you give it - and only accepts openstack resource objects. 

In the docs we should push people to only use this to chain multiple queries